### ISSUE TITLE
[information] OpenCL.dll is missing after the latest Windows 10 update

### ISSUE THREAD

**Issue by earthengine** [gcp/leela-zero#620]
For workaround solutions check

https://social.technet.microsoft.com/Forums/en-US/85680348-c2c4-40bc-9f39-9dcfeea331c0/windows-10-opencldll-error?forum=win10itprogeneral

(Note this is an old thread, but I hope some information listed there might help)

I have just experiencing this issue and it is unfortunate that it is the exact time Leela zero new version being released so people will feel like this is an issue of Leela zero, but it is not.

**Comment by gcp** [comment]
I don't see why this would be an issue as I ship a proper non-vendor specific OpenCL.dll?

**Comment by earthengine** [comment]
This is still an issue is you do not use an OpenCL.dll come from the official display driver (expecially from a different brend), I think. Also, missing one file might not be an issue but I don't know whether other files are also missing. But Let me see if it can solve it.

**Comment by earthengine** [comment]
It is a major problem in my system. After attempted to install the Nvidia driver again the display driver stopped working and it uses the "Microsoft basic driver" only, which means I lost access to the GPU.

**Comment by gcp** [comment]
>This is still an issue is you do not use an OpenCL.dll come from the official display driver (expecially from a different brend)

There is absolutely no need to use the OpenCL.dll from the official display driver, as long as it's a correct implementation of an ICD Loader. I ship the Khronos reference implementation which can correctly broker to multiple vendor specific drivers.

Whatever problem you are having is completely unrelated to the Leela distribution, which completely sidesteps the issue by bundling a proper OpenCL.dll. (After I explained you several times why that was needed, IIRC).

**Comment by gcp** [comment]
And yes, if the update removed your display driver nothing will work. That has got nothing whatsoever to do with OpenCL.dll. Because, AGAIN, we ship a proper implementation that is not vendor specific.

**Comment by earthengine** [comment]
OK, the second attempt to re-install the driver succeeded. Everything is now fine.

Issue timeline:

1. I installed a major windows update (a re-install of Windows actually)
2. When installation finished everything seems OK but Leela zero does not work - shows OpenCL.dll missing message
3. The display driver seems correct, but when running GeForce Experience it stucked at the splash screen forever
4. I kill the splash screen and download and install the latest Nvidia driver. It crashes on a access violation and after that the display driver being failed back to "Microsoft basic driver"
5. Install the driver the second time, all problems solved

This is actually saying, if I shipped another OpenCL.dll, it might or might not work but the OpenCL message will be at least different, which will make the problem being harder to address I think. In my case, the missing of OpenCL.dll indicates a major problem of the OS, and I need to fix this eventually.
